solids: # Listes des matrices de transformation homogènes des marqueurs ArUco dans le repère du solide associé
  - id: 0
    name: "base"
    aruco_markers:
      - id: 16
        transform:
          - [1.0, 0.0, 0.0, 0.1]
          - [0.0, 1.0, 0.0, 0.2] 
          - [0.0, 0.0, 1.0, 0.3]
          - [0.0, 0.0, 0.0, 1.0]
      - id: 17
        transform:
          - [0.866, -0.5, 0.0, 0.2]
          - [0.5, 0.866, 0.0, 0.3]
          - [0.0, 0.0, 1.0, 0.4]
          - [0.0, 0.0, 0.0, 1.0]
      - id: 18
        transform:
          - [0.707, -0.707, 0.0, 0.3]
          - [0.707, 0.707, 0.0, 0.4]
          - [0.0, 0.0, 1.0, 0.5]
          - [0.0, 0.0, 0.0, 1.0]
      - id: 19
        transform:
          - [0.5, -0.866, 0.0, 0.4]
          - [0.866, 0.5, 0.0, 0.5]
          - [0.0, 0.0, 1.0, 0.6]
          - [0.0, 0.0, 0.0, 1.0]
  - id: 1
    name: "arm_segment_1"
    aruco_markers:
      - id: 0
        transform:
          - [1.0, 0.0, 0.0, 0.5]
          - [0.0, 1.0, 0.0, 0.6]
          - [0.0, 0.0, 1.0, 0.7]
          - [0.0, 0.0, 0.0, 1.0]
      - id: 1
        transform:
          - [0.707, -0.707, 0.0, 0.1]
          - [0.707, 0.707, 0.0, 0.2]
          - [0.0, 0.0, 1.0, 0.3]
          - [0.0, 0.0, 0.0, 1.0]
      - id: 2
        transform:
          - [0.5, -0.866, 0.0, 0.3]
          - [0.866, 0.5, 0.0, 0.4]
          - [0.0, 0.0, 1.0, 0.5]
          - [0.0, 0.0, 0.0, 1.0]
      - id: 3
        transform:
          - [0.258, -0.966, 0.0, 0.4]
          - [0.966, 0.258, 0.0, 0.5]
          - [0.0, 0.0, 1.0, 0.6]
          - [0.0, 0.0, 0.0, 1.0]
  - id: 2
    name: "arm_segment_2"
    aruco_markers:
      - id: 4
        transform:
          - [1.0, 0.0, 0.0, 0.8]
          - [0.0, 1.0, 0.0, 0.9]
          - [0.0, 0.0, 1.0, 1.0]
          - [0.0, 0.0, 0.0, 1.0]
      - id: 5
        transform:
          - [0.866, -0.5, 0.0, 0.4]
          - [0.5, 0.866, 0.0, 0.5]
          - [0.0, 0.0, 1.0, 0.6]
          - [0.0, 0.0, 0.0, 1.0]
      - id: 6
        transform:
          - [0.707, -0.707, 0.0, 0.2]
          - [0.707, 0.707, 0.0, 0.3]
          - [0.0, 0.0, 1.0, 0.4]
          - [0.0, 0.0, 0.0, 1.0]
      - id: 7
        transform:
          - [0.5, -0.866, 0.0, 0.0]
          - [0.866, 0.5, 0.0, 0.1]
          - [0.0, 0.0, 1.0, 0.2]
          - [0.0, 0.0, 0.0, 1.0]
  - id: 3
    name : "arm_segment_3"
    aruco_markers:
      - id: 8
        transform:
          - [1.0, 0.0, 0.0, 1.1]
          - [0.0, 1.0, 0.0, 1.2]
          - [0.0, 0.0, 1.0, 1.3]
          - [0.0, 0.0, 0.0, 1.0]
      - id: 9
        transform:
          - [0.866, -0.5, 0.0, 0.7]
          - [0.5, 0.866, 0.0, 0.8]
          - [0.0, 0.0, 1.0, 0.9]
          - [0.0, 0.0, 0.0, 1.0]
      - id: 10
        transform:
          - [0.707, -0.707, 0.0, 0.5]
          - [0.707, 0.707, 0.0, 0.6]
          - [0.0, 0.0, 1.0, 0.7]
          - [0.0, 0.0, 0.0, 1.0]
      - id: 11
        transform:
          - [0.5, -0.866, 0.0, 0.3]
          - [0.866, 0.5, 0.0, 0.4]
          - [0.0, 0.0, 1.0, 0.5]
          - [0.0, 0.0, 0.0, 1.0]
  - id: 4
    name : "arm_segment_4"
    aruco_markers:
      - id: 12
        transform:
          - [1.0, 0.0, 0.0, 1.4]
          - [0.0, 1.0, 0.0, 1.5]
          - [0.0, 0.0, 1.0, 1.6]
          - [0.0, 0.0, 0.0, 1.0]
      - id: 13
        transform:
          - [0.866, -0.5, 0.0, 1.0]
          - [0.5, 0.866, 0.0, 1.1]
          - [0.0, 0.0, 1.0, 1.2]
          - [0.0, 0.0, 0.0, 1.0]
      - id: 14
        transform:
          - [0.707, -0.707, 0.0, 0.8]
          - [0.707, 0.707, 0.0, 0.9]
          - [0.0, 0.0, 1.0, 1.0]
          - [0.0, 0.0, 0.0, 1.0]
      - id: 15
        transform:
          - [0.5, -0.866, 0.0, 0.6]
          - [0.866, 0.5, 0.0, 0.7]
          - [0.0, 0.0, 1.0, 0.8]
          - [0.0, 0.0, 0.0, 1.0]
camera:
  - id: 0
    transform: # Matrice de transformation de la caméra dans le repère de la base
      - [1.0, 0.0, 0.0, 0.0]
      - [0.0, 1.0, 0.0, 0.0]
      - [0.0, 0.0, 1.0, 0.0]
      - [0.0, 0.0, 0.0, 1.0]